test/sol5/kyberNetworkProxy.js

```diff
@@ -578,6 +602,392 @@ contract('KyberNetworkProxy', function(accounts) {
+            let srcToken = tokens[1];
+            let dstToken = tokens[2];
+            let amountTwei = 1313;
+
+            tokenAmountForTrades = new BN(10000).mul(new BN(10).pow(tokenDecimals[1]));
+            await srcToken.transfer(taker, amountTwei);
+            await srcToken.transfer(generousNetwork.address, tokenAmountForTrades);
+            //get rate
+            let rate = await networkProxy.getExpectedRate(srcToken.address, dstToken.address, amountTwei);
+            await srcToken.approve(networkProxy.address, amountTwei, { from: taker });
+
+            //see trade reverts
+            await expectRevert(
+                networkProxy.trade(
+                    srcToken.address,
+                    amountTwei,
+                    dstToken.address,
+                    taker,
+                    maxDestAmt,
+                    rate.expectedRate,
+                    zeroAddress,
+                    { from: taker }
+                ), "srcAdd bad qty"
+            );
+            // change the amount and see trade success
+            await networkProxy.trade(
+                srcToken.address,
+                1312,
+                dstToken.address,
+                taker,
+                maxDestAmt,
+                rate.worstRate,
+                zeroAddress,
+                { from: taker }
+            );
+        });
+
+
+        it("trade revert if dest balance after is smaller than dest balance before", async() =>{
+            await networkProxy.setKyberNetwork(maliciousNetwork.address, { from: admin });
+            let srcToken = tokens[1];
+            let dstToken = tokens[2];
+            let amountTwei = 1000;
+            //get rate
+            let rate = await networkProxy.getExpectedRate(srcToken.address, dstToken.address, amountTwei);
+            dstQty = Helper.calcDstQty(amountTwei, tokenDecimals[1], tokenDecimals[2], rate.expectedRate);
+            await srcToken.transfer(taker, amountTwei);
+            await srcToken.approve(networkProxy.address, amountTwei, { from: taker });
+            await dstToken.transfer(destAddress, new BN(2200));
+            await dstToken.approve(maliciousNetwork.address, maxDestAmt, { from: destAddress});
+            await dstToken.approve(networkProxy.address, maxDestAmt, { from: destAddress});
+            //set my fee wei is greater than dstQty so maliciousNetwork will not send dest Token
+            await maliciousNetwork.setMyFeeWei(new BN(dstQty));
+            //see trade reverts due to maliciousNetwork reduce dest Amount 
+            await expectRevert(
+                networkProxy.trade(
+                    srcToken.address,
+                    amountTwei,
+                    dstToken.address,
+                    taker,
+                    maxDestAmt,
+                    rate.expectedRate,
+                    destAddress,
+                    { from: taker }
+                ), "destAdd bad qty"
```

Comment:
please update in proxy contract.
wrong amount in destination address.
And also here...

Reply:
updated in https://github.com/KyberNetwork/smart-contracts/pull/726/commits/e4d067f71b9400d43a678940a586bbb43991f02a